src/main/java/org/shanerx/tradeshop/itrade/IShopCreateEventListener.java

```diff
@@ -54,16 +54,16 @@ public void onSignChange(SignChangeEvent event) {
         if (!(event.getLine(0).equalsIgnoreCase(header))) {
             return;
         }
-
-        Block chest = findShopChest(s.getBlock());
-
+        
         if (!player.hasPermission(getCreateIPerm())) {
             failedSign(event, ShopType.ITRADE, Message.NO_TS_CREATE_PERMISSION);
             return;
         }
 
-        if (findShopChest(s.getBlock()) != null && getShopUsers(findShopChest(s.getBlock())).size() > 0) {
-            getShopOwners(s).stream().forEach(op -> {
+        Block chest = findShopChest(s.getBlock());
+        
+        if (chest != null && getShopOwners(chest).size() > 0) {
+            getShopOwners(chest).stream().forEach(op -> {
```

Comment:
Collection#stream().forEach() can be shortened to Collection#forEach(), worth considering for (micro-) performance & code tidiness reasons.